### ISSUE TITLE
Missing half the channels

### ISSUE THREAD

**Issue by ghost** [interlark/123tv-iptv#7]
All of a sudden 123tv-iptv is missing many of the channels. For example, all espn, fox sports etc. They are available on the 123tv site. Maybe the stream-id changed and the channel.json needs updating?

**Comment by Nostradamus1973** [comment]
Good morning,
Yeah, I'm experiencing the same thing, my installation is only loading 3 to 8 channels(depending on the day) out of 83 channels.
Is anyone else experiencing the same thing? Before something changed I used to load 17 out of 17 channels, now this.
Does anyone know how to fix this? Maybe a purge and a reload?
Thx

**Comment by interlark** [comment]
*123TV* isn't an easy thing, a lot of its channels uses *P2P* streaming, which isn't stable from time to time.
You can check `Keep all channels` checkbox or use `--keep-all-channels` argument to keep all the channels despite of their availability at the moment.

**Comment by ghost** [comment]
Some of the channels you can view at the website but you can't see them through your app. I looked at your channels.json file and the site-id matches to what I see on 123live site so I'm not sure why it doesn't play in the app.

**Comment by oatmealm** [comment]
I can see many channels prodcued in the xml guide, but are not visible from within Jellyfin... any ideas why this is happening? For exampe, MSNBC is not found even though the guide contains all the information about the channels and programs when I watch the source.

![image](https://user-images.githubusercontent.com/68159077/215259887-a2163a96-cfc7-4cee-af29-16f2121d4374.png)

**Comment by oatmealm** [comment]
BTW, I'm seeing this after enabling --keep-all-channels... when I play MSNBC:

```
09:56:12 :: INFO :: Refreshing auth key for channel MSNBC.
09:56:21 :: WARNING :: Channel "MSNBC" returned invalid playlist.
09:56:22 :: WARNING :: Channel "MSNBC" returned invalid playlist.
09:57:31 :: INFO :: Refreshing auth key for channel MSNBC.
09:57:39 :: WARNING :: Channel "MSNBC" returned invalid playlist.
```

**Comment by Nostradamus1973** [comment]
Well, I sorta asked that on the 20th and Andy answered it on the 28th.
The reason is Stability, the channels that aren't showing up are not stable, hence they are omitted.
MSNBC is one of those streams, it registers, however because it's unstable(at that point in time) it won't load; you could keep trying and see if further attemps produce watchable streams.
But, given the fact the USTVGO has given up it's ghost, I might be down to get unstable and load all the channels.

So, how would we go about using the --keep-all-channels argument?
So, with Ubuntu, I would just use the following:

sudo -E env "PATH=$PATH" 123tv-iptv install-service --keep-all-channels

@interlark Would that work?

**Comment by interlark** [comment]
@oatmealm, MSNBC is a P2P stream and it's not working, you can check it with your browser: https://123tv.live/watch/msnbc

There are regular channels and P2P channels.
P2P channels doesn't work for some reasons sometimes, due to lack of peers, I guess.
Regular channels works anytime, like [Comedy Central](https://123tv.live/watch/comedy-central/) or [Disney](https://123tv.live/watch/disney-channel/), etc.

[123tv-iptv](https://github.com/interlark/123tv-iptv) grabs only working channels at the moment, but as @Nostradamus1973 said, you can grab all of them (with option `--keep-all-channels`) whether they work or not, and hope they would work again in near future.

If you like to have only working channels in your playlist, just leave things as it is

**Comment by interlark** [comment]
I agree with you guys, [123tv](https://123tv.live/) is not that good as [ustvgo](http://ustvgo.tv/), but at least it has something to watch for free :)

**Comment by oatmealm** [comment]
Yes I can see it with `--keep-all-channels` indeed, and lots of  other channels, but as you say, it doesnt streem...

**Comment by oatmealm** [comment]
BTW are you aware of daddyhd? The site is horrible but seem to have a very good selection of channels, not just USA based. It looks like other than endless popups and popunders they're able to stream via google tv... I think,

**Comment by oatmealm** [comment]
Also https://ustvgo.live/msnbc seems to work

**Comment by netstrem** [comment]
@oatmealm I personally use uBlock Origin to keep me safe when using daddyhd and was able to encounter no ads. The site is pretty horrible otherwise, to the point where you can't even click to unmute the player unless you have an adblock enabled.

The streams seem to work pretty well, just unfortunately I haven't been able to get them to play outside of the site. https://freestreams-live1.tv/ seems to use the same CDN (webtv1.lol). Any idea why these streams only work in the web browser?
